packages/aws-cdk-lib/aws-cloudfront/lib/function.ts

```diff
@@ -31,12 +32,93 @@ export abstract class FunctionCode {
    * renders the function code
    */
   public abstract render(): string;
+
+  protected findReplace(code: string, instructions: FunctionCodeFindReplace[] | undefined): string {
+    if (!instructions?.length) {
+      return code;
+    }
+    function reducer(acc: string, inst: FunctionCodeFindReplace) {
+      const replacement = typeof inst.replace === 'string' ? inst.replace : Token.asString(inst.replace);
+      if (inst.all) {
+        // Use String.prototype.replaceAll once eslib is es2021+
+        // Simplification of https://github.com/es-shims/String.prototype.replaceAll/blob/main/implementation.js
+        const advanceBy = Math.max(1, inst.find.length);
+        const matchPositions: number[] = [];
+        for (
+          let position = acc.indexOf(inst.find);
+          position !== -1;
+          position = acc.indexOf(inst.find, position + advanceBy)
+        ) {
+          matchPositions.push(position);
+        }
+        let endOfLastMatch = 0;
+        let result = '';
+        for (const matchPos of matchPositions) {
+          result += acc.slice(endOfLastMatch, matchPos);
+          result += replacement;
+          endOfLastMatch = matchPos + inst.find.length;
+        }
+        if (endOfLastMatch < acc.length) {
+          result += acc.slice(endOfLastMatch);
+        }
+        return result;
+      } else {
+        return acc.replace(inst.find, replacement);
+      }
+    }
+    return instructions.reduce(reducer, code);
+  }
+
+}
+
+/**
+ * Options common to both inline code and code from an external file.
+ */
+export interface FunctionCodeOptions {
+  /**
+   * A set of instructions for the CDK to find text in the CloudFront function's code and replace it with other text.
+   *
+   * The CDK does each find/replace in order, with the output of one find/replace becomes the input to the next.
+   *
+   * @default - do not modify the code
+   */
+  readonly findReplace?: FunctionCodeFindReplace[];
+}
+
+/**
+ * An instruction for the CDK to find text in a CloudFront function's code and replacing it with other text.
+ *
+ * The CDK finds any match for the text. It is not restricted to whole identifiers, strings, or words.
+ * It is recommended to put a suffix and prefix in the find text, such as '%TEXT%' or '__TEXT__'.
+ */
+export interface FunctionCodeFindReplace {
+  /**
+   * The text to find.
+   */
+  readonly find: string;
+
+  /**
+   * The text to replace the find text with.
+   */
+  readonly replace: string | Token;
```

Comment:
Be careful here - deploy-time tokens will not be resolved with this, will they?

Reply:
I think they do. In the case of KVS ID, it's not known until the KVS is deployed.